### PR TITLE
Use keras application backbone for better result.

### PR DESCRIPTION
**Pull request by tanzhenyu**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/training/object_detection/pascal_voc/faster_rcnn.py [hunk 1]
```diff
@@ -64,9 +64,31 @@
 eval_ds = tfds.load("voc/2007", split="test", with_info=False)
 
 with strategy.scope():
-    backbone = keras_cv.models.ResNet50(
-        include_top=False, weights="imagenet", include_rescaling=True
-    ).as_backbone()
+    inputs = tf.keras.layers.Input(shape=image_size)
```
Comment (ianstenbit) [inline examples/training/object_detection/pascal_voc/faster_rcnn.py:67]: Let’s make this a helper function so it doesn’t crowd out the more relevant code here. (Same for the other script)

**Review by ianstenbit — APPROVED** [review]
LGTM if we put building the backbone into a helper method in both cases

File: examples/training/object_detection/pascal_voc/faster_rcnn.py (hunk 1 above)
Comment (tanzhenyu) [reply examples/training/object_detection/pascal_voc/faster_rcnn.py:67]: Actually the 2 scripts are using different levels of outputs from backbone, so we'd have to create a slightly more generic helper function to return different backbones in this case

**Pull request merged**